lambda/handler.go

```diff
@@ -41,19 +42,33 @@ func errorHandler(e error) lambdaHandler {
 }
 
 func validateArguments(handler reflect.Type) (bool, error) {
-	handlerTakesContext := false
-	if handler.NumIn() > 2 {
-		return false, fmt.Errorf("handlers may not take more than two arguments, but handler takes %d", handler.NumIn())
-	} else if handler.NumIn() > 0 {
+	switch handler.NumIn() {
+	case 0:
+		return false, nil
+	case 1:
 		contextType := reflect.TypeOf((*context.Context)(nil)).Elem()
 		argumentType := handler.In(0)
-		handlerTakesContext = argumentType.Implements(contextType)
-		if handler.NumIn() > 1 && !handlerTakesContext {
+		if argumentType.Kind() != reflect.Interface {
+			return false, nil
+		}
+		if !contextType.Implements(argumentType) {
+			return false, errors.New("the first argument is an interface, but it is not a Context")
+		}
+		if argumentType.NumMethod() == 0 {
```

Comment:
In this case, the handler has the following signature.

```go
func handler(v interface{}) {}

func main() { lambda.Start(handler) }
```

We have two choices to call the handler.
First is:

```go
ctx := context.TODO()
handler(ctx)
```

and, second is:

```go
v, _ := json.Unmarshal(data)
handler(v)
```

Calling the handler succeeds in either case, but we need to choose one of them.
Current implementation chooses the second.
So, to avoid breaking backward compatibility, we should the second here.

Reply:
Assuming the first argument does implement `valuer` (ie `contextType.implements(argumentType) == true`) wouldn't it always have at least 1 exported method? And if it did wouldn't this check always return false?